src/app/components/ContactsList.js

```diff
@@ -1,22 +1,20 @@
 import React, { useState, useEffect, useRef } from 'react';
 import PropTypes from 'prop-types';
-import { useContactGroups, useUser } from 'react-components';
+import { useContactGroups } from 'react-components';
 import { withRouter } from 'react-router';
 import { addPlus, getInitial } from 'proton-shared/lib/helpers/string';
 import List from 'react-virtualized/dist/commonjs/List';
 import AutoSizer from 'react-virtualized/dist/commonjs/AutoSizer';
 
-import ContactGroupIcon from './ContactGroupIcon';
 import ItemCheckbox from './ItemCheckbox';
-import { extract } from '../helpers/merge';
+import ContactGroupIcon from './ContactGroupIcon';
+import { extractMergeable } from '../helpers/merge';
 import { c } from 'ttag';
 import MergeRow from './MergeRow';
 
-const ContactsList = ({ contacts, onCheck, history, contactID, location }) => {
-    const [{ hasPaidMail }] = useUser();
-    const emails = extract(contacts);
-    const duplicates = Object.values(emails).reduce((acc, arr) => acc + arr.length, 0);
-    const canMerge = duplicates > 0;
+const ContactsList = ({ contacts, onCheck, user, userKeysList, loadingUserKeys, history, contactID, location }) => {
+    const mergeableContacts = extractMergeable(contacts);
```

Comment:
```suggestion
    const mergeableContacts = useMemo(() => extractMergeable(contacts), [contacts]);
```

To not have perf issues with contacts lists that are large

Reply:
Thanks for the suggestion!